pxblue/react-native-component-library
# Add Deprecation warnings for props

#### Changes proposed in this Pull Request:
- Add deprecation warnings (console.warn) for properties that will be going away or changing
- Add the new props and a conversion from old -> new to facilitate users making the changes before upgrading to v6
- 

#### To Test:
- Run the showcase demo
- You should be presented with a bunch of YellowBox warnings
- Make some of the necessary changes (e.g., changing `IconClass` to `icon`) and you should see the warning count go down.
